algorithm.go

```diff
@@ -92,100 +67,39 @@ func (a Algorithm) String() string {
 		// As stated in RFC 8152 8.2, only the pure EdDSA version is used for
 		// COSE.
 		return "EdDSA"
+	default:
+		return "unknown algorithm value " + strconv.Itoa(int(a))
```

Comment:
this seems more like an error message than a friendly string. What about `return "UnknownAlgorithm(" + strconv.Itoa(int(a)) + ")"`

Reply:
This has the same output as the `crypto` package. See https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/crypto/crypto.go;l=64

Reply:
`crypto.Hash` is a bad example, as it can't be extended with custom hashed. Notice that all its method panics if the hash value is outside the allowed range. The `Hash.String()` method is the only one that don't panic, and maybe that's why it returns an error-like string.

But I might be overthinking, feel free to discard this comment.